# Preprocessing and Data Encoding

In this document, we preprocess the dataset to make sure we only have clean data. 

Towards the end, we are going to perform categorical data encoding.

## 1 Preprocessing for Clean Data

### 1.1 Loading the data

First, we load the data from our [dataset](https://docs.google.com/spreadsheets/d/1jzNEJWQMvfK6aJeG28JSXecrLZfjeFOCYp-gtW0Q7LQ/edit#gid=1456408654) which we have made available online. 

We display the last three records as shown in the output to ensure that we have obtained all data properly.

In [84]:
# Load datasets
import pandas as pd
import gspread
import numpy as np

sa = gspread.service_account(filename="group-28-dataset-7233caedfe09.json")
sheet = sa.open("Dataset - Group 28")
work_sheet = sheet.worksheet("Fixed Data")

df = pd.DataFrame(work_sheet.get_all_values())

new_header = df.iloc[0] 
df = df[1:152] 
df.columns = new_header 
df = df.replace(r'^\s*$', np.nan, regex=True)

df.tail(3)


0        ID            Timestamp   
149  28-149  18/04/2023 23:47:41  \
150  28-150  18/04/2023 23:54:57   
151  28-151  18/04/2023 23:59:24   

0                                            Tweet URL Group   
149  https://twitter.com/fenum9/status/137914056758...    28  \
150  https://twitter.com/Wengsterrr_/status/1371652...    28   
151  https://twitter.com/AdrianSBanal/status/141888...    28   

0           Collector Category   
149  Pardiñas, Miguel     HLTH  \
150  Pardiñas, Miguel     HLTH   
151  Pardiñas, Miguel     HLTH   

0                                                Topic   
149  COVID-19 vaccines contain microchip, magnetic ...  \
150  COVID-19 vaccines contain microchip, magnetic ...   
151  COVID-19 vaccines contain microchip, magnetic ...   

0                     Keywords Account handle   Account name  ...   
149  "covid" "vaccine" "metal"        @fenum9  h♤yd$n 🧙‍♂️ 𓅃  ...  \
150           "bakuna" "tubig"   @Wengsterrr_     𝕃𝕦𝕒𝕟𝕟 𝕃𝕚 👅  ...   
151           "bakuna" "tubig"  @AdrianSBanal           Nabs  ...   

0   Quote Tweets Views Rating   
149            0   NaN    NaN  \
150            0   NaN    NaN   
151            0   NaN    NaN   

0                                            Reasoning   
149  Tells that nurse on duty during vaccine appoin...  \
150  Complains why cases went up, and points that p...   
151  Complains that vaccine might not be working, a...   

0                   Remarks Thread/Tweet Language Add columns here   
149             No location               Foreign              NaN  \
150  Location might be fake              Filipino              NaN   
151                     NaN              Filipino              NaN   

0   Add columns here Reviewer Review  
149              NaN      NaN    NaN  
150              NaN      NaN    NaN  
151              NaN      NaN    NaN  

[3 rows x 35 columns]

Shown here are all the current features of the dataset. 

This may include features with missing values.

In [85]:
print("List all current features:")
df.columns

List all current features:


Index(['ID', 'Timestamp', 'Tweet URL', 'Group', 'Collector', 'Category',
       'Topic', 'Keywords', 'Account handle', 'Account name', 'Account bio',
       'Account type', 'Joined', 'Following', 'Followers', 'Location', 'Tweet',
       'Tweet Translated', 'Tweet Type', 'Date posted', 'Screenshot',
       'Content type', 'Likes', 'Replies', 'Retweets', 'Quote Tweets', 'Views',
       'Rating', 'Reasoning', 'Remarks', 'Thread/Tweet Language',
       'Add columns here', 'Add columns here', 'Reviewer', 'Review'],
      dtype='object', name=0)

In [86]:
print("Display the number of records and features:")
df.shape

Display the number of records and features:


(151, 35)

### 1.2 Checking for Missing Values

Now, we check which features has no missing values.

From, here we can see that there are missing value on Account Bio, Location, Screenshot, etc.

In [87]:
print("Check for missing values:")
df.isnull().sum()

Check for missing values:


0
ID                         0
Timestamp                  0
Tweet URL                  0
Group                      0
Collector                  0
Category                   0
Topic                      0
Keywords                   0
Account handle             0
Account name               0
Account bio               25
Account type               1
Joined                     0
Following                  0
Followers                  0
Location                  52
Tweet                      0
Tweet Translated           0
Tweet Type                 0
Date posted                0
Screenshot               101
Content type               0
Likes                      0
Replies                    0
Retweets                   0
Quote Tweets              49
Views                    151
Rating                   150
Reasoning                  0
Remarks                   32
Thread/Tweet Language      0
Add columns here         151
Add columns here         151
Reviewer                 151
Review      

Upon checking, the collectors deemed that the only necessary features for this project are as follows:

> `['ID', 'Timestamp', 'Tweet URL', 'Group', 'Keywords',
    'Account handle', 'Account name', 'Account bio', 'Account type', 
    'Tweet', 'Tweet Translated', 'Tweet Type', 'Date posted',
    'Content type', 'Reasoning', 'Thread/Tweet Language',]`

From the previous output, we can verify that all of these features have no missing value.

In [88]:
print("Dataset summary:")
df_clean = df[['ID', 'Timestamp', 'Tweet URL', 'Group', 'Keywords', 
         'Account handle', 'Account name', 'Account bio', 'Account type', 
         'Tweet', 'Tweet Translated', 'Tweet Type', 'Date posted',
         'Content type', 'Reasoning', 'Thread/Tweet Language',]]
df_clean.describe().style

Dataset summary:


The table shows a simple summary of our clean dataset. Some notable observations are as follows: 
- ID, Tweet URL, Tweet, and Tweet Translated have unique values.
- 82 out of 151 tweets are of Emotional content type.
- There are 78 Foreign Tweets. This also implies that the remaining 73 tweets are Filipino Tweets.
- Ven Cuenca has four tweets on this dataset.

### 1.3 Ensuring Formatting Cosistency

Words

## 2 Categorical Data Encoding

### 2.1 One Hot Encoding

Words

### 2.2 Heat Map (Feature Correlation)

Words